narwhals/_duckdb/dataframe.py

```diff
@@ -184,7 +184,12 @@ def select(
         *exprs: DuckDBExpr,
     ) -> Self:
         selection = (val.alias(name) for name, val in evaluate_exprs(self, *exprs))
-        return self._with_native(self.native.select(*selection))
+        try:
+            return self._with_native(self.native.select(*selection))
+        except duckdb.BinderException as e:
+            if "this column cannot be referenced before it is defined" in str(e):
+                raise ColumnNotFoundError.from_available_column_names(self.columns) from e
+            raise
```

Comment:
should we do this for `simple_select` and `aggregate` too?

i think in polars we have a `catch_exception` utility, could/should we make something like that?

Reply:
Sorry @MarcoGorelli I didn't have time to look at this yet :(

I just saw that `catch_polars_exception` it is way cleaner! I'll update later tonight